blockindex/sgd_indexer.go

```diff
@@ -285,14 +281,10 @@ func (sgd *sgdRegistry) StartHeight() uint64 {
 
 // PutBlock puts a block into SGDIndexer
 func (sgd *sgdRegistry) PutBlock(ctx context.Context, blk *block.Block) error {
-	tipHeight, err := sgd.Height()
+	expectHeight, err := sgd.expectHeight()
```

Comment:
nit: also do this for contract staking indexer? to keep it consistent

Reply:
sorry i mean wrapping the L284-294 into a subfunc